// @aws-amplify/ui-react/dist/Field-B8Bw-DBK.js
'use strict';

var React = require('react');
var core = require('@aws-amplify/core');
var auth = require('aws-amplify/auth');
var uiReactCore = require('@aws-amplify/ui-react-core');
var ui = require('@aws-amplify/ui');

function _interopNamespace(e) {
    if (e && e.__esModule) return e;
    var n = Object.create(null);
    if (e) {
        Object.keys(e).forEach(function (k) {
            if (k !== 'default') {
                var d = Object.getOwnPropertyDescriptor(e, k);
                Object.defineProperty(n, k, d.get ? d : {
                    enumerable: true,
                    get: function () { return e[k]; }
                });
            }
        });
    }
    n["default"] = e;
    return Object.freeze(n);
}

var React__namespace = /*#__PURE__*/_interopNamespace(React);

/**
 * Amplify Auth React hook
 * @internal
 */
const useAuth = () => {
    const [result, setResult] = React__namespace.useState({
        error: undefined,
        isLoading: true,
        user: undefined,
    });
    /**
     * Hub events like `tokenRefresh` will not give back the user object.
     * This util will be used to get current user after those events.
     */
    const fetchCurrentUser = React__namespace.useCallback(async () => {
        setResult((prevResult) => ({ ...prevResult, isLoading: true }));
        try {
            const user = await auth.getCurrentUser();
            setResult({ user, isLoading: false });
        }
        catch (e) {
            const error = e;
            setResult({ error, isLoading: false });
        }
    }, []);
    const handleAuth = React__namespace.useCallback(({ payload }) => {
        switch (payload.event) {
            // success events
            case 'signedIn':
            case 'signUp':
            case 'autoSignIn': {
                setResult({ user: payload.data, isLoading: false });
                break;
            }
            case 'signedOut': {
                setResult({ user: undefined, isLoading: false });
                break;
            }
            // failure events
            case 'tokenRefresh_failure':
            case 'signIn_failure': {
                setResult({ error: payload.data, isLoading: false });
                break;
            }
            case 'autoSignIn_failure': {
                // autoSignIn just returns error message. Wrap it to an Error object
                setResult({ error: new Error(payload.message), isLoading: false });
                break;
            }
            // events that need another fetch
            case 'tokenRefresh': {
                fetchCurrentUser();
                break;
            }
        }
    }, [fetchCurrentUser]);
    React__namespace.useEffect(() => {
        const unsubscribe = core.Hub.listen('auth', handleAuth, 'useAuth');
        fetchCurrentUser(); // on init, see if user is already logged in
        return unsubscribe;
    }, [handleAuth, fetchCurrentUser]);
    return {
        ...result,
        /** @deprecated Fetch is handled automatically, do not use this directly */
        fetch: fetchCurrentUser,
    };
};

const ThemeContext = React__namespace.createContext({
    theme: ui.createTheme(),
    colorMode: undefined,
});

/**
 * Get current Theme object value from Amplify context.
 * Returns a default theme if context is not available
 */
const getThemeFromContext = (context) => {
    if (typeof context === 'undefined' || typeof context.theme === 'undefined') {
        return ui.createTheme();
    }
    return context.theme;
};
/**
 * [📖 Docs](https://ui.docs.amplify.aws/react/theming)
 */
const useTheme = () => {
    const context = React__namespace.useContext(ThemeContext);
    return getThemeFromContext(context);
};
/**
 * Internal use only
 */
const useColorMode = () => {
    const context = React__namespace.useContext(ThemeContext);
    return context.colorMode;
};

// For internal use, no need to export
const ComponentText = {
    Alert: {
        dismissButtonLabel: 'Dismiss alert',
    },
    Autocomplete: {
        emptyText: 'No options found',
        loadingText: 'Loading options...',
    },
    Collection: {
        searchButtonLabel: 'Search',
        searchNoResultsFound: 'No results found',
    },
    Fields: {
        clearButtonLabel: 'Clear input',
    },
    Message: {
        dismissLabel: 'Dismiss message',
    },
    PaginationItem: {
        currentPageLabel: 'Page',
        nextLabel: 'Go to next page',
        pageLabel: 'Go to page',
        previousLabel: 'Go to previous page',
    },
    PhoneNumberField: {
        countryCodeLabel: 'Country code',
    },
    SearchField: {
        searchButtonLabel: 'Search',
    },
    PasswordField: {
        passwordIsHidden: 'Password is hidden',
        passwordIsShown: 'Password is shown',
        showPassword: 'Show password',
    },
    StepperField: {
        increaseButtonLabel: 'Increase to',
        decreaseButtonLabel: 'Decrease to',
    },
};
const stylePropsToThemeKeys = {
    backgroundColor: 'colors',
    borderColor: 'colors',
    borderWidth: 'borderWidths',
    color: 'colors',
    borderRadius: 'radii',
    fontSize: 'fontSizes',
    fontWeight: 'fontWeights',
    fontFamily: 'fonts',
    lineHeight: 'lineHeights',
    opacity: 'opacities',
    boxShadow: 'shadows',
    transform: 'transforms',
    left: 'space',
    right: 'space',
    top: 'space',
    bottom: 'space',
    height: 'space',
    width: 'space',
    letterSpacing: 'space',
    margin: 'space',
    marginBlock: 'space',
    marginBlockEnd: 'space',
    marginBlockStart: 'space',
    marginInline: 'space',
    marginInlineEnd: 'space',
    marginInlineStart: 'space',
    marginLeft: 'space',
    marginRight: 'space',
    marginTop: 'space',
    marginBottom: 'space',
    maxHeight: 'space',
    maxWidth: 'space',
    minHeight: 'space',
    minWidth: 'space',
    padding: 'space',
    paddingBlock: 'space',
    paddingBlockEnd: 'space',
    paddingBlockStart: 'space',
    paddingInline: 'space',
    paddingInlineEnd: 'space',
    paddingInlineStart: 'space',
    paddingLeft: 'space',
    paddingRight: 'space',
    paddingTop: 'space',
    paddingBottom: 'space',
    gap: 'space',
    columnGap: 'space',
    rowGap: 'space',
};
// key name
const ESCAPE_KEY = 'Escape';
const ENTER_KEY = 'Enter';
const ARROW_UP = 'ArrowUp';
const ARROW_DOWN = 'ArrowDown';

const isThemeStylePropKey = (key) => {
    return key in stylePropsToThemeKeys;
};

const strHasLength = (str) => typeof str === 'string' && str.length > 0;
const isEmptyString = (value) => typeof value === 'string' && value.length === 0;
const isNullOrEmptyString = (value) => value == null || isEmptyString(value);
/**
 * Create a consecutive integer array from start value to end value.
 * @param start start value
 * @param end end value
 * @returns an integer array with elements from start to end consecutively
 */
const getConsecutiveIntArray = (start, end) => {
    const length = end - start + 1;
    return Array.from({ length }, (_, idx) => idx + start);
};
/**
 * TS helper function to make using Object.keys more typesafe
 */
const objectKeys = (obj) => {
    return Object.keys(obj);
};
const getCSSVariableIfValueIsThemeKey = (propKey, value, tokens) => {
    if (typeof value !== 'string') {
        return value;
    }
    // For shorthand properties like `padding` which can accept 1, 2, 3, or 4 values
    // run this function on each value. This would not work on CSS shorthands that
    // mix types, like border which is a composite of borderWidth, borderStyle, and
    // borderColor.
    if (value.includes(' ')) {
        return value
            .split(' ')
            .map((val) => getCSSVariableIfValueIsThemeKey(propKey, val, tokens))
            .join(' ');
    }
    if (isThemeStylePropKey(propKey)) {
        const path = value.split('.');
        const tokenKey = stylePropsToThemeKeys[propKey];
        let tokenProps = tokens[tokenKey];
        for (let i = 0; i < path.length; i++) {
            if (tokenProps) {
                // overwrite tokenProps with next nested value of tokenProps
                tokenProps = tokenProps[path[i]];
                continue;
            }
            break;
        }
        return ui.isDesignToken(tokenProps)
            ? `var(--${ui.cssNameTransform({
                path: [stylePropsToThemeKeys[propKey], ...path],
            })})`
            : value;
    }
    return value;
};

// Inspiration for getMediaQueries and useBreakpoint
// comes from https://github.com/iiroj/use-breakpoint/
const getMediaQueries = ({ breakpoints }) => {
    const sortedBreakpoints = objectKeys(breakpoints).sort((a, b) => breakpoints[b] - breakpoints[a]);
    return sortedBreakpoints.map((breakpoint, index) => {
        let query = '';
        const minWidth = breakpoints[breakpoint];
        const nextBreakpoint = sortedBreakpoints[index - 1];
        const maxWidth = nextBreakpoint ? breakpoints[nextBreakpoint] - 1 : null;
        if (minWidth >= 0) {
            query = `(min-width: ${minWidth}px)`;
        }
        if (maxWidth !== null) {
            if (query) {
                query += ' and ';
            }
            query += `(max-width: ${maxWidth}px)`;
        }
        return {
            breakpoint,
            query,
            maxWidth,
            minWidth,
        };
    });
};

// Inspiration for getMediaQueries and useBreakpoint
// comes from https://github.com/iiroj/use-breakpoint/
const useIsomorphicEffect = typeof window === 'undefined' ? React__namespace.useEffect : React__namespace.useLayoutEffect;
const useBreakpoint = ({ breakpoints, defaultBreakpoint, }) => {
    const supportMatchMedia = typeof window !== 'undefined' && typeof window.matchMedia !== 'undefined';
    const matchMedia = supportMatchMedia ? window.matchMedia : null;
    const mediaQueries = React__namespace.useMemo(() => getMediaQueries({ breakpoints }), [breakpoints]);
    const [breakpoint, setBreakpoint] = React__namespace.useState(defaultBreakpoint);
    const updateBreakpoint = React__namespace.useCallback((matches, breakpoint) => {
        if (matches) {
            setBreakpoint(breakpoint);
        }
    }, [setBreakpoint]);
    useIsomorphicEffect(() => {
        if (!matchMedia)
            return;
        const unsubscribeList = mediaQueries.map(({ query, breakpoint }) => {
            const queryList = matchMedia(query);
            updateBreakpoint(queryList.matches, breakpoint);
            const handleMediaChange = (event) => {
                if (event.matches) {
                    setBreakpoint(breakpoint);
                }
            };
            queryList.addEventListener('change', handleMediaChange);
            return () => queryList.removeEventListener('change', handleMediaChange);
        });
        return () => {
            unsubscribeList.forEach((unsubscribe) => unsubscribe());
        };
    }, [breakpoints, setBreakpoint, matchMedia, mediaQueries]);
    /** Print a nice debug value for React Devtools */
    React__namespace.useDebugValue(breakpoint, (breakpoint) => breakpoint);
    return breakpoint;
};

/**
 * Logs a deprecation warning `message` to the console.
 */
const useDeprecationWarning = ({ message, shouldWarn: _shouldWarn, }) => {
    const shouldWarn = _shouldWarn &&
        // show message on builds without Node `process` polyfill
        // or with process.env.NODE_ENV not production
        (typeof process === 'undefined' ||
            (process && process.env.NODE_ENV !== 'production'));
    uiReactCore.useDeprecationWarning({ message, shouldWarn });
};

/**
 * @internal May be removed in a future version
 * Maps from component style props to React `style` props
 * Note: Primarily needed to map from component style props that don't match CSS Properties directly
 * such as wrap => flexWrap and direction => flexDirection
 */
const ComponentPropsToStylePropsMap = {
    alignContent: 'alignContent',
    alignItems: 'alignItems',
    alignSelf: 'alignSelf',
    area: 'gridArea',
    aspectRatio: 'aspectRatio',
    autoColumns: 'gridAutoColumns',
    autoFlow: 'gridAutoFlow',
    autoRows: 'gridAutoRows',
    backgroundColor: 'backgroundColor',
    backgroundImage: 'backgroundImage',
    basis: 'flexBasis',
    border: 'border',
    borderRadius: 'borderRadius',
    borderColor: 'borderColor',
    borderWidth: 'borderWidth',
    borderStyle: 'borderStyle',
    bottom: 'bottom',
    boxShadow: 'boxShadow',
    color: 'color',
    column: 'gridColumn',
    columnEnd: 'gridColumnEnd',
    columnGap: 'columnGap',
    columnSpan: 'gridColumn',
    columnStart: 'gridColumnStart',
    direction: 'flexDirection',
    display: 'display',
    flex: 'flex',
    fontFamily: 'fontFamily',
    fontSize: 'fontSize',
    fontStyle: 'fontStyle',
    fontWeight: 'fontWeight',
    gap: 'gap',
    grow: 'flexGrow',
    height: 'height',
    justifyContent: 'justifyContent',
    left: 'left',
    letterSpacing: 'letterSpacing',
    lineHeight: 'lineHeight',
    margin: 'margin',
    marginBlock: 'marginBlock',
    marginBlockEnd: 'marginBlockEnd',
    marginBlockStart: 'marginBlockStart',
    marginBottom: 'marginBlockEnd',
    marginInline: 'marginInline',
    marginInlineEnd: 'marginInlineEnd',
    marginInlineStart: 'marginInlineStart',
    marginLeft: 'marginInlineStart',
    marginRight: 'marginInlineEnd',
    marginTop: 'marginBlockStart',
    maxHeight: 'maxHeight',
    maxWidth: 'maxWidth',
    minHeight: 'minHeight',
    minWidth: 'minWidth',
    objectFit: 'objectFit',
    objectPosition: 'objectPosition',
    opacity: 'opacity',
    order: 'order',
    overflow: 'overflow',
    padding: 'padding',
    paddingBlock: 'paddingBlock',
    paddingBlockEnd: 'paddingBlockEnd',
    paddingBlockStart: 'paddingBlockStart',
    paddingBottom: 'paddingBlockEnd',
    paddingInline: 'paddingInline',
    paddingInlineEnd: 'paddingInlineEnd',
    paddingInlineStart: 'paddingInlineStart',
    paddingLeft: 'paddingInlineStart',
    paddingRight: 'paddingInlineEnd',
    paddingTop: 'paddingBlockStart',
    position: 'position',
    resize: 'resize',
    right: 'right',
    row: 'gridRow',
    rowEnd: 'gridRowEnd',
    rowGap: 'rowGap',
    rowSpan: 'gridRow',
    rowStart: 'gridRowStart',
    shrink: 'flexShrink',
    templateAreas: 'gridTemplateAreas',
    templateColumns: 'gridTemplateColumns',
    templateRows: 'gridTemplateRows',
    textAlign: 'textAlign',
    textDecoration: 'textDecoration',
    textTransform: 'textTransform',
    top: 'top',
    transform: 'transform',
    transformOrigin: 'transformOrigin',
    width: 'width',
    whiteSpace: 'whiteSpace',
    wrap: 'flexWrap',
};
/**
 * @internal May be removed in a future version
 */
const ComponentPropsToStylePropsMapKeys = Object.keys(ComponentPropsToStylePropsMap);

const getClosestValueByBreakpoint = ({ breakpoint, breakpoints, values, }) => {
    const value = values[breakpoint];
    // Check if breakpoint exists in values
    if (value !== undefined) {
        return value;
    }
    // Otherwise use a lower breakpoint value
    const breakpointsDesc = objectKeys(breakpoints).sort((a, b) => breakpoints[b] - breakpoints[a]);
    const lowerBreakpoints = breakpointsDesc.slice(breakpointsDesc.indexOf(breakpoint));
    for (const breakpoint of lowerBreakpoints) {
        // Check if breakpoint exists in values
        const value = values[breakpoint];
        if (value !== undefined) {
            return value;
        }
    }
    return null;
};
/**
 * This takes an object and will return an object that only has the
 * breakpoint keys
 * @param obj
 * @returns
 */
const valueObjToBreakpoints = (obj) => {
    return objectKeys(obj).reduce((acc, key) => key in ui.defaultTheme.breakpoints.values
        ? { ...acc, [key]: obj[key] }
        : acc, {});
};
const getValueAtCurrentBreakpoint = ({ breakpoint, breakpoints, values, }) => {
    let breakpointCompatValues = {};
    const breakpointsAscending = objectKeys(breakpoints).sort((a, b) => breakpoints[a] - breakpoints[b]);
    if (Array.isArray(values)) {
        values.forEach((value, index) => {
            breakpointCompatValues[breakpointsAscending[index]] = value;
        });
    }
    else if (typeof values === 'object') {
        breakpointCompatValues = valueObjToBreakpoints(values);
    }
    return getClosestValueByBreakpoint({
        breakpoint,
        breakpoints,
        values: breakpointCompatValues,
    });
};

/**
 * This takes an unknown value, which could be a:
 * - design token: `color={tokens.colors.font.primary}`
 * - string, which could be a:
 *   - theme key: `color='font.primary'`
 *   - plain style: `color='red'`
 * - or a number: `padding={10}`
 * and returns the appropriate and resolved value
 */
const getStyleValue = ({ value, propKey, tokens, }) => {
    if (ui.isDesignToken(value)) {
        return value.toString();
    }
    if (ui.isString(value)) {
        return ui.isString(propKey)
            ? getCSSVariableIfValueIsThemeKey(propKey, value, tokens)
            : value;
    }
    return null;
};

const isSpanPrimitiveValue = (spanValue) => {
    return (spanValue === 'auto' ||
        (typeof spanValue === 'number' && !isNaN(spanValue)) ||
        (typeof spanValue === 'string' && !isNaN(parseFloat(spanValue))));
};
const getGridSpan = (spanValue) => {
    return spanValue === 'auto' ? 'auto' : `span ${spanValue}`;
};
const convertGridSpan = (spanValue) => {
    // PropertyType
    if (isSpanPrimitiveValue(spanValue)) {
        return getGridSpan(spanValue);
    }
    // PropertyType[]
    if (Array.isArray(spanValue)) {
        return spanValue.map((value) => getGridSpan(value));
    }
    // ResponsiveObject<PropertyType>
    if (typeof spanValue === 'object' && spanValue != null) {
        return Object.entries(spanValue).reduce((acc, [key, value]) => ({ ...acc, [key]: getGridSpan(value) }), {});
    }
    return null;
};
/**
 * Transforms style props to another target prop
 * where the original is a simpler API than the target.
 * This function will remove the original prop and
 * replace target prop values with calculated
 * E.g. rowSpan => row, columnSpan => column
 */
const useTransformStyleProps = (props) => {
    const { rowSpan, columnSpan, row, column, ...rest } = props;
    const { rowFromSpanValue, columnFromSpanValue } = React__namespace.useMemo(() => {
        return {
            rowFromSpanValue: convertGridSpan(rowSpan),
            columnFromSpanValue: convertGridSpan(columnSpan),
        };
    }, [rowSpan, columnSpan]);
    return {
        row: !isNullOrEmptyString(row) ? row : rowFromSpanValue,
        column: !isNullOrEmptyString(column) ? column : columnFromSpanValue,
        ...rest,
    };
};
const isComponentStyleProp = (key) => {
    return key in ComponentPropsToStylePropsMap;
};
/**
 * Convert style props to CSS variables for React style prop
 * Note: Will filter out undefined, null, and empty string prop values
 */
const convertStylePropsToStyleObj = ({ props = {}, style = {}, breakpoint, breakpoints, tokens, }) => {
    const nonStyleProps = {};
    Object.keys(props)
        .filter((propKey) => props[propKey] !== null)
        .forEach((propKey) => {
        if (isComponentStyleProp(propKey)) {
            const values = props[propKey];
            if (values === null || values === undefined || isEmptyString(values))
                return;
            const reactStyleProp = ComponentPropsToStylePropsMap[propKey];
            // short circuit the style prop here if it is a string or design token
            // so we don't have to call getValueAtCurrentBreakpoint every time
            let value = '';
            if (ui.isDesignToken(values)) {
                value = values.toString();
            }
            else if (typeof values === 'string') {
                value = getCSSVariableIfValueIsThemeKey(propKey, values, tokens);
            }
            else if (typeof values === 'number') {
                value = values;
            }
            else if (typeof values === 'object') {
                // here values should be a responsive array or object
                value = getStyleValue({
                    propKey,
                    tokens,
                    value: getValueAtCurrentBreakpoint({
                        values,
                        breakpoint,
                        breakpoints,
                    }),
                });
            }
            style = {
                ...style,
                [reactStyleProp]: value,
            };
        }
        else if (typeof props[propKey] !== 'undefined') {
            nonStyleProps[propKey] = props[propKey];
        }
    });
    return { propStyles: style, nonStyleProps };
};
const useStyles = (props, style) => {
    const { breakpoints: { values: breakpoints, defaultBreakpoint }, tokens, } = useTheme();
    const breakpoint = useBreakpoint({
        breakpoints,
        defaultBreakpoint,
    });
    const propStyles = useTransformStyleProps(props);
    return React__namespace.useMemo(() => convertStylePropsToStyleObj({
        props: propStyles,
        style,
        breakpoint,
        breakpoints,
        tokens,
    }), [propStyles, style, breakpoints, breakpoint, tokens]);
};

/**
 * Updates the return type for primitives wrapped in `React.forwardRef` to
 * `React.ReactElement`. In React 18 the return type of `React.ExoticComponent`
 * was changed from `React.ReactElement` to `React.ReactNode`, which breaks
 * clients using React 16 and 17.
 *
 * @param primitive UI Primitive to be wrapped with `React.forwardRef`
 * @returns ForwaredRef wrapped UI Primitive
 */
const primitiveWithForwardRef = (primitive) => React__namespace.forwardRef(primitive);

const ViewPrimitive = ({ as = 'div', children, testId, ariaLabel, isDisabled, style, inert, ...rest }, ref) => {
    const { propStyles, nonStyleProps } = useStyles(rest, style);
    return React__namespace.createElement(as, {
        'aria-label': ariaLabel,
        'data-testid': testId,
        disabled: isDisabled,
        ref,
        inert: inert ? '' : null,
        style: propStyles,
        ...nonStyleProps,
    }, children);
};
/**
 * [📖 Docs](https://ui.docs.amplify.aws/react/components/view)
 */
const View = primitiveWithForwardRef(ViewPrimitive);
View.displayName = 'View';

const defaultViewBox = { minX: 0, minY: 0, width: 24, height: 24 };
const IconPrimitive = ({ className, 
// as can be used to render other icon react components too
as = 'svg', fill = 'currentColor', pathData, viewBox = defaultViewBox, children, paths, ...rest }, ref) => {
    const minX = viewBox.minX ?? defaultViewBox.minX;
    const minY = viewBox.minY ?? defaultViewBox.minY;
    const width = viewBox.width ?? defaultViewBox.width;
    const height = viewBox.height ?? defaultViewBox.height;
    // An icon can be drawn in 3 ways:
    // 1. Pass it children which should be valid SVG elements
    // 2. Pass an array of path-like objects to `paths` prop
    // 3. Supply `pathData` for a simple icons
    let _children;
    if (children) {
        _children = children;
    }
    if (paths) {
        _children = paths.map((path, index) => React__namespace.createElement("path", { ...path, key: index }));
    }
    if (pathData) {
        _children = React__namespace.createElement("path", { d: pathData, fill: fill });
    }
    return (React__namespace.createElement(View, { as: as, className: ui.classNames(ui.ComponentClassName.Icon, className), ref: ref, viewBox: `${minX} ${minY} ${width} ${height}`, ...rest }, _children));
};
/**
 * [📖 Docs](https://ui.docs.amplify.aws/react/components/icon)
 */
const Icon = primitiveWithForwardRef(IconPrimitive);
Icon.displayName = 'Icon';

const IconsContext = React__namespace.createContext({});

function useIcons(component) {
    const context = React__namespace.useContext(IconsContext);
    if (component && context) {
        return context[component];
    }
    return undefined;
}

/**
 * @internal For internal Amplify UI use only. May be removed in a future release.
 */
const IconAdd = (props) => {
    const { className, ...rest } = props;
    return (React__namespace.createElement(View, { as: "span", width: "1em", height: "1em", className: ui.classNames(ui.ComponentClassName.Icon, className), ...rest },
        React__namespace.createElement("svg", { width: "24", height: "24", viewBox: "0 0 24 24", fill: "none", xmlns: "http://www.w3.org/2000/svg" },
            React__namespace.createElement("path", { d: "M19 13H13V19H11V13H5V11H11V5H13V11H19V13Z", fill: "currentColor" }))));
};

/**
 * @internal For internal Amplify UI use only. May be removed in a future release.
 */
const IconCheckCircleFill = (props) => {
    const { className, ...rest } = props;
    return (React__namespace.createElement(View, { as: "span", width: "1em", height: "1em", className: ui.classNames(ui.ComponentClassName.Icon, className), ...rest },
        React__namespace.createElement("svg", { width: "54", height: "54", viewBox: "0 0 54 54", fill: "none", xmlns: "http://www.w3.org/2000/svg" },
            React__namespace.createElement("path", { d: "M22.9333 38.9333L41.7333 20.1333L38 16.4L22.9333 31.4667L15.3333 23.8667L11.6 27.6L22.9333 38.9333ZM26.6667 53.3333C22.9778 53.3333 19.5111 52.6333 16.2667 51.2333C13.0222 49.8333 10.2 47.9333 7.8 45.5333C5.4 43.1333 3.5 40.3111 2.1 37.0667C0.7 33.8222 0 30.3556 0 26.6667C0 22.9778 0.7 19.5111 2.1 16.2667C3.5 13.0222 5.4 10.2 7.8 7.8C10.2 5.4 13.0222 3.5 16.2667 2.1C19.5111 0.7 22.9778 0 26.6667 0C30.3556 0 33.8222 0.7 37.0667 2.1C40.3111 3.5 43.1333 5.4 45.5333 7.8C47.9333 10.2 49.8333 13.0222 51.2333 16.2667C52.6333 19.5111 53.3333 22.9778 53.3333 26.6667C53.3333 30.3556 52.6333 33.8222 51.2333 37.0667C49.8333 40.3111 47.9333 43.1333 45.5333 45.5333C43.1333 47.9333 40.3111 49.8333 37.0667 51.2333C33.8222 52.6333 30.3556 53.3333 26.6667 53.3333Z", fill: "currentColor" }))));
};

/**
 * @internal For internal Amplify UI use only. May be removed in a future release.
 */
const IconCheckCircle = (props) => {
    const { className, ...rest } = props;
    return (React__namespace.createElement(View, { as: "span", width: "1em", height: "1em", className: ui.classNames(ui.ComponentClassName.Icon, className), ...rest },
        React__namespace.createElement("svg", { width: "24", height: "24", viewBox: "0 0 24 24", fill: "none", xmlns: "http://www.w3.org/2000/svg" },
            React__namespace.createElement("path", { d: "M12 2C6.48 2 2 6.48 2 12C2 17.52 6.48 22 12 22C17.52 22 22 17.52 22 12C22 6.48 17.52 2 12 2ZM12 20C7.59 20 4 16.41 4 12C4 7.59 7.59 4 12 4C16.41 4 20 7.59 20 12C20 16.41 16.41 20 12 20ZM16.59 7.58L10 14.17L7.41 11.59L6 13L10 17L18 9L16.59 7.58Z", fill: "currentColor" }))));
};

/**
 * @internal For internal Amplify UI use only. May be removed in a future release.
 */
const IconCheck = (props) => {
    const { className, ...rest } = props;
    return (React__namespace.createElement(View, { as: "span", width: "1em", height: "1em", className: ui.classNames(ui.ComponentClassName.Icon, className), ...rest },
        React__namespace.createElement("svg", { width: "24", height: "24", viewBox: "0 0 24 24", fill: "none", xmlns: "http://www.w3.org/2000/svg" },
            React__namespace.createElement("path", { d: "M9.00016 16.1698L4.83016 11.9998L3.41016 13.4098L9.00016 18.9998L21.0002 6.99984L19.5902 5.58984L9.00016 16.1698Z", fill: "currentColor" }))));
};

/**
 * @internal For internal Amplify UI use only. May be removed in a future release.
 */
const IconChevronLeft = (props) => {
    const { className, ...rest } = props;
    return (React__namespace.createElement(View, { as: "span", width: "1em", height: "1em", className: ui.classNames(ui.ComponentClassName.Icon, className), ...rest },
        React__namespace.createElement("svg", { width: "24", height: "24", viewBox: "0 0 24 24", fill: "none", xmlns: "http://www.w3.org/2000/svg" },
            React__namespace.createElement("path", { d: "M15.41 7.41L14 6L8 12L14 18L15.41 16.59L10.83 12L15.41 7.41Z", fill: "currentColor" }))));
};

/**
 * @internal For internal Amplify UI use only. May be removed in a future release.
 */
const IconChevronRight = (props) => {
    const { className, ...rest } = props;
    return (React__namespace.createElement(View, { as: "span", width: "1em", height: "1em", className: ui.classNames(ui.ComponentClassName.Icon, className), ...rest },
        React__namespace.createElement("svg", { width: "24", height: "24", viewBox: "0 0 24 24", fill: "none", xmlns: "http://www.w3.org/2000/svg" },
            React__namespace.createElement("path", { d: "M9.99984 6L8.58984 7.41L13.1698 12L8.58984 16.59L9.99984 18L15.9998 12L9.99984 6Z", fill: "currentColor" }))));
};

/**
 * @internal For internal Amplify UI use only. May be removed in a future release.
 */
const IconClose = (props) => {
    const { className, size, ...rest } = props;
    return (React__namespace.createElement(View, { as: "span", width: size ?? '1em', height: size ?? '1em', className: ui.classNames(ui.ComponentClassName.Icon, className), ...rest },
        React__namespace.createElement("svg", { width: "24", height: "24", viewBox: "0 0 24 24", fill: "none", xmlns: "http://www.w3.org/2000/svg", style: size ? { width: size, height: size } : undefined },
            React__namespace.createElement("path", { d: "M19 6.41L17.59 5L12 10.59L6.41 5L5 6.41L10.59 12L5 17.59L6.41 19L12 13.41L17.59 19L19 17.59L13.41 12L19 6.41Z", fill: "currentColor" }))));
};

/**
 * @internal For internal Amplify UI use only. May be removed in a future release.
 */
const IconError = (props) => {
    const { className, ...rest } = props;
    return (React__namespace.createElement(View, { as: "span", width: "1em", height: "1em", className: ui.classNames(ui.ComponentClassName.Icon, className), ...rest },
        React__namespace.createElement("svg", { width: "24", height: "24", viewBox: "0 0 24 24", fill: "none", xmlns: "http://www.w3.org/2000/svg" },
            React__namespace.createElement("path", { d: "M12 2C6.48 2 2 6.48 2 12C2 17.52 6.48 22 12 22C17.52 22 22 17.52 22 12C22 6.48 17.52 2 12 2ZM13 17H11V15H13V17ZM13 13H11V7H13V13Z", fill: "currentColor" }))));
};

/**
 * @internal For internal Amplify UI use only. May be removed in a future release.
 */
const IconExpandMore = (props) => {
    const { className, ...rest } = props;
    return (React__namespace.createElement(View, { as: "span", width: "1em", height: "1em", className: ui.classNames(ui.ComponentClassName.Icon, className), ...rest },
        React__namespace.createElement("svg", { width: "24", height: "24", viewBox: "0 0 24 24", fill: "none", xmlns: "http://www.w3.org/2000/svg" },
            React__namespace.createElement("path", { d: "M16.59 8.58984L12 13.1698L7.41 8.58984L6 9.99984L12 15.9998L18 9.99984L16.59 8.58984Z", fill: "currentColor" }))));
};

/**
 * @internal For internal Amplify UI use only. May be removed in a future release.
 */
const IconIndeterminate = (props) => {
    const { className, ...rest } = props;
    return (React__namespace.createElement(View, { as: "span", width: "1em", height: "1em", className: ui.classNames(ui.ComponentClassName.Icon, className), ...rest },
        React__namespace.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", height: "24", width: "24", viewBox: "0 0 24 24" },
            React__namespace.createElement("line", { x1: "4", x2: "20", y1: "12", y2: "12", stroke: "currentColor", strokeWidth: "3" }))));
};

/**
 * @internal For internal Amplify UI use only. May be removed in a future release.
 */
const IconInfo = (props) => {
    const { className, ...rest } = props;
    return (React__namespace.createElement(View, { as: "span", width: "1em", height: "1em", className: ui.classNames(ui.ComponentClassName.Icon, className), ...rest },
        React__namespace.createElement("svg", { width: "24", height: "24", viewBox: "0 0 24 24", fill: "none", xmlns: "http://www.w3.org/2000/svg" },
            React__namespace.createElement("path", { d: "M11 7H13V9H11V7ZM11 11H13V17H11V11ZM12 2C6.48 2 2 6.48 2 12C2 17.52 6.48 22 12 22C17.52 22 22 17.52 22 12C22 6.48 17.52 2 12 2ZM12 20C7.59 20 4 16.41 4 12C4 7.59 7.59 4 12 4C16.41 4 20 7.59 20 12C20 16.41 16.41 20 12 20Z", fill: "currentColor" }))));
};

/**
 * @internal For internal Amplify UI use only. May be removed in a future release.
 */
const IconMenu = (props) => {
    const { className, ...rest } = props;
    return (React__namespace.createElement(View, { as: "span", width: "1em", height: "1em", className: ui.classNames(ui.ComponentClassName.Icon, className), ...rest },
        React__namespace.createElement("svg", { width: "24", height: "24", viewBox: "0 0 24 24", fill: "none", xmlns: "http://www.w3.org/2000/svg" },
            React__namespace.createElement("path", { d: "M3 18H21V16H3V18ZM3 13H21V11H3V13ZM3 6V8H21V6H3Z", fill: "currentColor" }))));
};

/**
 * @internal For internal Amplify UI use only. May be removed in a future release.
 */
const IconRemove = (props) => {
    const { className, ...rest } = props;
    return (React__namespace.createElement(View, { as: "span", width: "1em", height: "1em", className: ui.classNames(ui.ComponentClassName.Icon, className), ...rest },
        React__namespace.createElement("svg", { width: "24", height: "24", viewBox: "0 0 24 24", fill: "none", xmlns: "http://www.w3.org/2000/svg" },
            React__namespace.createElement("path", { d: "M19 13H5V11H19V13Z", fill: "currentColor" }))));
};

/**
 * @internal For internal Amplify UI use only. May be removed in a future release.
 */
const IconSearch = (props) => {
    const { className, ...rest } = props;
    return (React__namespace.createElement(View, { as: "span", width: "1em", height: "1em", className: ui.classNames(ui.ComponentClassName.Icon, className), ...rest },
        React__namespace.createElement("svg", { width: "24", height: "24", viewBox: "0 0 24 24", fill: "none", xmlns: "http://www.w3.org/2000/svg" },
            React__namespace.createElement("path", { d: "M15.5 14H14.71L14.43 13.73C15.41 12.59 16 11.11 16 9.5C16 5.91 13.09 3 9.5 3C5.91 3 3 5.91 3 9.5C3 13.09 5.91 16 9.5 16C11.11 16 12.59 15.41 13.73 14.43L14 14.71V15.5L19 20.49L20.49 19L15.5 14V14ZM9.5 14C7.01 14 5 11.99 5 9.5C5 7.01 7.01 5 9.5 5C11.99 5 14 7.01 14 9.5C14 11.99 11.99 14 9.5 14Z", fill: "currentColor" }))));
};

/**
 * @internal For internal Amplify UI use only. May be removed in a future release.
 */
const IconStar = (props) => {
    const { className, ...rest } = props;
    return (React__namespace.createElement(View, { as: "span", width: "1em", height: "1em", className: ui.classNames(ui.ComponentClassName.Icon, className), ...rest },
        React__namespace.createElement("svg", { width: "24", height: "24", viewBox: "0 0 24 24", fill: "none", xmlns: "http://www.w3.org/2000/svg" },
            React__namespace.createElement("path", { d: "M12 17.27L18.18 21L16.54 13.97L22 9.24L14.81 8.63L12 2L9.19 8.63L2 9.24L7.46 13.97L5.82 21L12 17.27Z", fill: "currentColor" }))));
};

/**
 * @internal For internal Amplify UI use only. May be removed in a future release.
 */
const IconUser = (props) => {
    const { className, ...rest } = props;
    return (React__namespace.createElement(View, { as: "span", width: "1em", height: "1em", className: ui.classNames(ui.ComponentClassName.Icon, className), ...rest },
        React__namespace.createElement("svg", { width: "24", height: "24", viewBox: "0 0 24 24", fill: "none", xmlns: "http://www.w3.org/2000/svg" },
            React__namespace.createElement("path", { d: "M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z", fill: "currentColor" }))));
};

/**
 * @internal For internal Amplify UI use only. May be removed in a future release.
 */
const IconVisibilityOff = (props) => {
    const { className, ...rest } = props;
    return (React__namespace.createElement(View, { as: "span", width: "1em", height: "1em", className: ui.classNames(ui.ComponentClassName.Icon, className), ...rest },
        React__namespace.createElement("svg", { width: "24", height: "24", viewBox: "0 0 24 24", fill: "none", xmlns: "http://www.w3.org/2000/svg" },
            React__namespace.createElement("path", { d: "M12 6.0002C15.79 6.0002 19.17 8.1302 20.82 11.5002C20.23 12.7202 19.4 13.7702 18.41 14.6202L19.82 16.0302C21.21 14.8002 22.31 13.2602 23 11.5002C21.27 7.1102 17 4.0002 12 4.0002C10.73 4.0002 9.51 4.2002 8.36 4.5702L10.01 6.2202C10.66 6.0902 11.32 6.0002 12 6.0002ZM10.93 7.14019L13 9.2102C13.57 9.4602 14.03 9.9202 14.28 10.4902L16.35 12.5602C16.43 12.2202 16.49 11.8602 16.49 11.4902C16.5 9.0102 14.48 7.0002 12 7.0002C11.63 7.0002 11.28 7.05019 10.93 7.14019ZM2.01 3.8702L4.69 6.5502C3.06 7.8302 1.77 9.5302 1 11.5002C2.73 15.8902 7 19.0002 12 19.0002C13.52 19.0002 14.98 18.7102 16.32 18.1802L19.74 21.6002L21.15 20.1902L3.42 2.4502L2.01 3.8702ZM9.51 11.3702L12.12 13.9802C12.08 13.9902 12.04 14.0002 12 14.0002C10.62 14.0002 9.5 12.8802 9.5 11.5002C9.5 11.4502 9.51 11.4202 9.51 11.3702V11.3702ZM6.11 7.97019L7.86 9.7202C7.63 10.2702 7.5 10.8702 7.5 11.5002C7.5 13.9802 9.52 16.0002 12 16.0002C12.63 16.0002 13.23 15.8702 13.77 15.6402L14.75 16.6202C13.87 16.8602 12.95 17.0002 12 17.0002C8.21 17.0002 4.83 14.8702 3.18 11.5002C3.88 10.0702 4.9 8.89019 6.11 7.97019Z", fill: "currentColor" }))));
};

/**
 * @internal For internal Amplify UI use only. May be removed in a future release.
 */
const IconVisibility = (props) => {
    const { className, ...rest } = props;
    return (React__namespace.createElement(View, { as: "span", width: "1em", height: "1em", className: ui.classNames(ui.ComponentClassName.Icon, className), ...rest },
        React__namespace.createElement("svg", { width: "24", height: "24", viewBox: "0 0 24 24", fill: "none", xmlns: "http://www.w3.org/2000/svg" },
            React__namespace.createElement("path", { d: "M12 6C15.79 6 19.17 8.13 20.82 11.5C19.17 14.87 15.79 17 12 17C8.21 17 4.83 14.87 3.18 11.5C4.83 8.13 8.21 6 12 6ZM12 4C7 4 2.73 7.11 1 11.5C2.73 15.89 7 19 12 19C17 19 21.27 15.89 23 11.5C21.27 7.11 17 4 12 4ZM12 9C13.38 9 14.5 10.12 14.5 11.5C14.5 12.88 13.38 14 12 14C10.62 14 9.5 12.88 9.5 11.5C9.5 10.12 10.62 9 12 9ZM12 7C9.52 7 7.5 9.02 7.5 11.5C7.5 13.98 9.52 16 12 16C14.48 16 16.5 13.98 16.5 11.5C16.5 9.02 14.48 7 12 7Z", fill: "currentColor" }))));
};

/**
 * @internal For internal Amplify UI use only. May be removed in a future release.
 */
const IconWarning = (props) => {
    const { className, ...rest } = props;
    return (React__namespace.createElement(View, { as: "span", width: "1em", height: "1em", className: ui.classNames(ui.ComponentClassName.Icon, className), ...rest },
        React__namespace.createElement("svg", { width: "24", height: "24", viewBox: "0 0 24 24", fill: "none", xmlns: "http://www.w3.org/2000/svg" },
            React__namespace.createElement("path", { d: "M1 21H23L12 2L1 21ZM13 18H11V16H13V18ZM13 14H11V10H13V14Z", fill: "currentColor" }))));
};

/**
 * @internal For internal Amplify UI use only. May be removed in a future release.
 */
const IconPasskey = (props) => {
    const { className, ...rest } = props;
    return (React__namespace.createElement(View, { as: "span", width: "1em", height: "1em", className: ui.classNames(ui.ComponentClassName.Icon, className), ...rest },
        React__namespace.createElement("svg", { width: "195", height: "128", viewBox: "0 0 195 128", fill: "none", xmlns: "http://www.w3.org/2000/svg" },
            React__namespace.createElement("circle", { cx: "60", cy: "36", r: "31", stroke: "#00404D", strokeWidth: "2" }),
            React__namespace.createElement("path", { d: "M60 69C84.4313 69 105.396 83.8504 114.357 105.021C116.774 110.729 115.382 116.168 111.798 120.23C108.191 124.319 102.358 127 96 127H24C17.6416 127 11.8094 124.319 8.20215 120.23C4.61836 116.168 3.22626 110.729 5.64258 105.021C14.6036 83.8504 35.5687 69 60 69Z", stroke: "#00404D", strokeWidth: "2" }),
            React__namespace.createElement("circle", { cx: "122", cy: "26", r: "25", fill: "#E9F9FC", stroke: "#00404D", strokeWidth: "2" }),
            React__namespace.createElement("mask", { id: "path-4-inside-1_4178_10171", fill: "white" },
                React__namespace.createElement("path", { d: "M121.951 13.927C124.887 13.927 127.658 14.5348 130.262 15.7503C132.866 16.9658 135.04 18.7222 136.786 21.0197C136.98 21.2601 137.042 21.4738 136.973 21.6608C136.903 21.8478 136.786 22.0081 136.619 22.1417C136.453 22.2753 136.259 22.3354 136.038 22.322C135.816 22.3086 135.622 22.1951 135.456 21.9814C133.932 19.8977 131.972 18.3015 129.576 17.1928C127.18 16.0842 124.638 15.5299 121.951 15.5299C119.264 15.5299 116.743 16.0842 114.388 17.1928C112.034 18.3015 110.081 19.8977 108.529 21.9814C108.363 22.2218 108.169 22.3554 107.947 22.3821C107.726 22.4088 107.532 22.3554 107.366 22.2218C107.172 22.0883 107.054 21.9213 107.012 21.7209C106.971 21.5206 107.033 21.3135 107.199 21.0998C108.917 18.8291 111.071 17.066 113.661 15.8104C116.251 14.5548 119.015 13.927 121.951 13.927ZM121.951 17.6937C125.691 17.6937 128.904 18.8959 131.591 21.3002C134.279 23.7045 135.622 26.6831 135.622 30.2361C135.622 31.5719 135.13 32.6872 134.147 33.5821C133.164 34.477 131.965 34.9245 130.553 34.9245C129.14 34.9245 127.928 34.477 126.917 33.5821C125.906 32.6872 125.4 31.5719 125.4 30.2361C125.4 29.3546 125.061 28.6132 124.382 28.0122C123.703 27.4111 122.893 27.1106 121.951 27.1106C121.009 27.1106 120.199 27.4111 119.52 28.0122C118.841 28.6132 118.502 29.3546 118.502 30.2361C118.502 32.8274 119.299 34.9913 120.891 36.7277C122.484 38.4642 124.541 39.6797 127.062 40.3742C127.311 40.4544 127.478 40.588 127.561 40.775C127.644 40.962 127.658 41.1623 127.602 41.376C127.547 41.563 127.436 41.7233 127.27 41.8569C127.104 41.9905 126.896 42.0305 126.647 41.9771C123.766 41.2825 121.411 39.9001 119.582 37.8297C117.754 35.7593 116.84 33.2281 116.84 30.2361C116.84 28.9004 117.339 27.7784 118.336 26.8701C119.333 25.9618 120.538 25.5077 121.951 25.5077C123.364 25.5077 124.569 25.9618 125.566 26.8701C126.563 27.7784 127.062 28.9004 127.062 30.2361C127.062 31.1177 127.408 31.859 128.101 32.4601C128.794 33.0612 129.611 33.3617 130.553 33.3617C131.495 33.3617 132.298 33.0612 132.963 32.4601C133.628 31.859 133.96 31.1177 133.96 30.2361C133.96 27.1373 132.783 24.5326 130.428 22.4222C128.073 20.3117 125.261 19.2565 121.993 19.2565C118.724 19.2565 115.912 20.3117 113.557 22.4222C111.202 24.5326 110.025 27.1239 110.025 30.1961C110.025 30.8372 110.087 31.6386 110.212 32.6004C110.337 33.5621 110.635 34.6841 111.106 35.9664C111.189 36.2068 111.182 36.4205 111.085 36.6075C110.988 36.7945 110.828 36.9281 110.607 37.0082C110.385 37.0884 110.171 37.0817 109.963 36.9882C109.755 36.8947 109.61 36.7411 109.526 36.5274C109.111 35.4855 108.813 34.4503 108.633 33.4218C108.453 32.3933 108.363 31.3314 108.363 30.2361C108.363 26.6831 109.7 23.7045 112.373 21.3002C115.046 18.8959 118.239 17.6937 121.951 17.6937ZM121.951 10C123.724 10 125.455 10.207 127.145 10.6211C128.835 11.0352 130.47 11.6296 132.049 12.4043C132.298 12.5379 132.443 12.6982 132.485 12.8852C132.526 13.0722 132.506 13.2592 132.423 13.4462C132.339 13.6332 132.201 13.7801 132.007 13.8869C131.813 13.9938 131.578 13.9804 131.301 13.8469C129.832 13.1256 128.316 12.5713 126.75 12.1839C125.185 11.7965 123.585 11.6029 121.951 11.6029C120.344 11.6029 118.765 11.7832 117.214 12.1438C115.663 12.5045 114.18 13.0722 112.768 13.8469C112.546 13.9804 112.324 14.0138 112.103 13.947C111.881 13.8803 111.715 13.74 111.604 13.5263C111.493 13.3126 111.466 13.1189 111.521 12.9453C111.576 12.7716 111.715 12.618 111.937 12.4844C113.488 11.683 115.109 11.0686 116.798 10.6411C118.488 10.2137 120.206 10 121.951 10ZM121.951 21.5807C124.527 21.5807 126.744 22.4155 128.6 24.0852C130.456 25.7548 131.384 27.8051 131.384 30.2361C131.384 30.4766 131.308 30.6702 131.155 30.8172C131.003 30.9641 130.802 31.0376 130.553 31.0376C130.331 31.0376 130.137 30.9641 129.971 30.8172C129.805 30.6702 129.722 30.4766 129.722 30.2361C129.722 28.2326 128.953 26.5562 127.415 25.2072C125.878 23.8581 124.056 23.1835 121.951 23.1835C119.846 23.1835 118.038 23.8581 116.528 25.2072C115.018 26.5562 114.264 28.2326 114.264 30.2361C114.264 32.4 114.651 34.2366 115.427 35.746C116.203 37.2553 117.339 38.7714 118.834 40.2941C119.001 40.4544 119.084 40.6414 119.084 40.8551C119.084 41.0688 119.001 41.2558 118.834 41.4161C118.668 41.5764 118.474 41.6565 118.253 41.6565C118.031 41.6565 117.837 41.5764 117.671 41.4161C116.037 39.7598 114.783 38.0701 113.91 36.347C113.038 34.624 112.601 32.587 112.601 30.2361C112.601 27.8051 113.516 25.7548 115.344 24.0852C117.172 22.4155 119.375 21.5807 121.951 21.5807ZM121.909 29.4347C122.159 29.4347 122.36 29.5148 122.512 29.6751C122.664 29.8354 122.741 30.0224 122.741 30.2361C122.741 32.2397 123.488 33.8826 124.984 35.1649C126.48 36.4472 128.226 37.0884 130.22 37.0884C130.386 37.0884 130.622 37.075 130.927 37.0483C131.231 37.0216 131.55 36.9815 131.882 36.9281C132.132 36.8747 132.346 36.9081 132.526 37.0283C132.706 37.1485 132.824 37.3288 132.88 37.5692C132.935 37.7829 132.893 37.9699 132.755 38.1302C132.616 38.2905 132.436 38.3974 132.215 38.4508C131.716 38.5844 131.28 38.6578 130.906 38.6712C130.532 38.6846 130.303 38.6912 130.22 38.6912C127.755 38.6912 125.615 37.8898 123.8 36.2869C121.986 34.6841 121.078 32.6671 121.078 30.2361C121.078 30.0224 121.155 29.8354 121.307 29.6751C121.459 29.5148 121.66 29.4347 121.909 29.4347Z" })),
            React__namespace.createElement("path", { d: "M121.951 13.927C124.887 13.927 127.658 14.5348 130.262 15.7503C132.866 16.9658 135.04 18.7222 136.786 21.0197C136.98 21.2601 137.042 21.4738 136.973 21.6608C136.903 21.8478 136.786 22.0081 136.619 22.1417C136.453 22.2753 136.259 22.3354 136.038 22.322C135.816 22.3086 135.622 22.1951 135.456 21.9814C133.932 19.8977 131.972 18.3015 129.576 17.1928C127.18 16.0842 124.638 15.5299 121.951 15.5299C119.264 15.5299 116.743 16.0842 114.388 17.1928C112.034 18.3015 110.081 19.8977 108.529 21.9814C108.363 22.2218 108.169 22.3554 107.947 22.3821C107.726 22.4088 107.532 22.3554 107.366 22.2218C107.172 22.0883 107.054 21.9213 107.012 21.7209C106.971 21.5206 107.033 21.3135 107.199 21.0998C108.917 18.8291 111.071 17.066 113.661 15.8104C116.251 14.5548 119.015 13.927 121.951 13.927ZM121.951 17.6937C125.691 17.6937 128.904 18.8959 131.591 21.3002C134.279 23.7045 135.622 26.6831 135.622 30.2361C135.622 31.5719 135.13 32.6872 134.147 33.5821C133.164 34.477 131.965 34.9245 130.553 34.9245C129.14 34.9245 127.928 34.477 126.917 33.5821C125.906 32.6872 125.4 31.5719 125.4 30.2361C125.4 29.3546 125.061 28.6132 124.382 28.0122C123.703 27.4111 122.893 27.1106 121.951 27.1106C121.009 27.1106 120.199 27.4111 119.52 28.0122C118.841 28.6132 118.502 29.3546 118.502 30.2361C118.502 32.8274 119.299 34.9913 120.891 36.7277C122.484 38.4642 124.541 39.6797 127.062 40.3742C127.311 40.4544 127.478 40.588 127.561 40.775C127.644 40.962 127.658 41.1623 127.602 41.376C127.547 41.563 127.436 41.7233 127.27 41.8569C127.104 41.9905 126.896 42.0305 126.647 41.9771C123.766 41.2825 121.411 39.9001 119.582 37.8297C117.754 35.7593 116.84 33.2281 116.84 30.2361C116.84 28.9004 117.339 27.7784 118.336 26.8701C119.333 25.9618 120.538 25.5077 121.951 25.5077C123.364 25.5077 124.569 25.9618 125.566 26.8701C126.563 27.7784 127.062 28.9004 127.062 30.2361C127.062 31.1177 127.408 31.859 128.101 32.4601C128.794 33.0612 129.611 33.3617 130.553 33.3617C131.495 33.3617 132.298 33.0612 132.963 32.4601C133.628 31.859 133.96 31.1177 133.96 30.2361C133.96 27.1373 132.783 24.5326 130.428 22.4222C128.073 20.3117 125.261 19.2565 121.993 19.2565C118.724 19.2565 115.912 20.3117 113.557 22.4222C111.202 24.5326 110.025 27.1239 110.025 30.1961C110.025 30.8372 110.087 31.6386 110.212 32.6004C110.337 33.5621 110.635 34.6841 111.106 35.9664C111.189 36.2068 111.182 36.4205 111.085 36.6075C110.988 36.7945 110.828 36.9281 110.607 37.0082C110.385 37.0884 110.171 37.0817 109.963 36.9882C109.755 36.8947 109.61 36.7411 109.526 36.5274C109.111 35.4855 108.813 34.4503 108.633 33.4218C108.453 32.3933 108.363 31.3314 108.363 30.2361C108.363 26.6831 109.7 23.7045 112.373 21.3002C115.046 18.8959 118.239 17.6937 121.951 17.6937ZM121.951 10C123.724 10 125.455 10.207 127.145 10.6211C128.835 11.0352 130.47 11.6296 132.049 12.4043C132.298 12.5379 132.443 12.6982 132.485 12.8852C132.526 13.0722 132.506 13.2592 132.423 13.4462C132.339 13.6332 132.201 13.7801 132.007 13.8869C131.813 13.9938 131.578 13.9804 131.301 13.8469C129.832 13.1256 128.316 12.5713 126.75 12.1839C125.185 11.7965 123.585 11.6029 121.951 11.6029C120.344 11.6029 118.765 11.7832 117.214 12.1438C115.663 12.5045 114.18 13.0722 112.768 13.8469C112.546 13.9804 112.324 14.0138 112.103 13.947C111.881 13.8803 111.715 13.74 111.604 13.5263C111.493 13.3126 111.466 13.1189 111.521 12.9453C111.576 12.7716 111.715 12.618 111.937 12.4844C113.488 11.683 115.109 11.0686 116.798 10.6411C118.488 10.2137 120.206 10 121.951 10ZM121.951 21.5807C124.527 21.5807 126.744 22.4155 128.6 24.0852C130.456 25.7548 131.384 27.8051 131.384 30.2361C131.384 30.4766 131.308 30.6702 131.155 30.8172C131.003 30.9641 130.802 31.0376 130.553 31.0376C130.331 31.0376 130.137 30.9641 129.971 30.8172C129.805 30.6702 129.722 30.4766 129.722 30.2361C129.722 28.2326 128.953 26.5562 127.415 25.2072C125.878 23.8581 124.056 23.1835 121.951 23.1835C119.846 23.1835 118.038 23.8581 116.528 25.2072C115.018 26.5562 114.264 28.2326 114.264 30.2361C114.264 32.4 114.651 34.2366 115.427 35.746C116.203 37.2553 117.339 38.7714 118.834 40.2941C119.001 40.4544 119.084 40.6414 119.084 40.8551C119.084 41.0688 119.001 41.2558 118.834 41.4161C118.668 41.5764 118.474 41.6565 118.253 41.6565C118.031 41.6565 117.837 41.5764 117.671 41.4161C116.037 39.7598 114.783 38.0701 113.91 36.347C113.038 34.624 112.601 32.587 112.601 30.2361C112.601 27.8051 113.516 25.7548 115.344 24.0852C117.172 22.4155 119.375 21.5807 121.951 21.5807ZM121.909 29.4347C122.159 29.4347 122.36 29.5148 122.512 29.6751C122.664 29.8354 122.741 30.0224 122.741 30.2361C122.741 32.2397 123.488 33.8826 124.984 35.1649C126.48 36.4472 128.226 37.0884 130.22 37.0884C130.386 37.0884 130.622 37.075 130.927 37.0483C131.231 37.0216 131.55 36.9815 131.882 36.9281C132.132 36.8747 132.346 36.9081 132.526 37.0283C132.706 37.1485 132.824 37.3288 132.88 37.5692C132.935 37.7829 132.893 37.9699 132.755 38.1302C132.616 38.2905 132.436 38.3974 132.215 38.4508C131.716 38.5844 131.28 38.6578 130.906 38.6712C130.532 38.6846 130.303 38.6912 130.22 38.6912C127.755 38.6912 125.615 37.8898 123.8 36.2869C121.986 34.6841 121.078 32.6671 121.078 30.2361C121.078 30.0224 121.155 29.8354 121.307 29.6751C121.459 29.5148 121.66 29.4347 121.909 29.4347Z", fill: "#00404D", stroke: "#00404D", strokeWidth: "4", mask: "url(#path-4-inside-1_4178_10171)" }),
            React__namespace.createElement("circle", { cx: "167.5", cy: "49.5", r: "19.5", fill: "#E9F9FC", stroke: "#00404D", strokeWidth: "2" }),
            React__namespace.createElement("path", { d: "M167.89 43.65L161.26 54.18H167.89", stroke: "#00404D", strokeWidth: "2" }),
            React__namespace.createElement("path", { d: "M167.5 60.42C173.316 60.42 178.03 55.7056 178.03 49.89V48.72", stroke: "#00404D", strokeWidth: "2" }),
            React__namespace.createElement("circle", { cx: "173.74", cy: "43.26", r: "1", fill: "#00404D", stroke: "#00404D", strokeWidth: "1.12" }),
            React__namespace.createElement("circle", { cx: "160.48", cy: "43.26", r: "1", fill: "#00404D", stroke: "#00404D", strokeWidth: "1.12" }),
            React__namespace.createElement("path", { fillRule: "evenodd", clipRule: "evenodd", d: "M104 112C116.129 112 126.457 104.288 130.347 93.5H147.5L155.5 101.5L168 89L180.5 101.5L193 89L180.5 76.5H130.984C127.705 64.6779 116.866 56 104 56C88.536 56 76 68.536 76 84C76 99.464 88.536 112 104 112ZM104 98C111.732 98 118 91.732 118 84C118 76.268 111.732 70 104 70C96.268 70 90 76.268 90 84C90 91.732 96.268 98 104 98Z", fill: "#BCECF5" }),
            React__namespace.createElement("path", { d: "M130.347 93.5V92.5H129.645L129.407 93.1608L130.347 93.5ZM147.5 93.5L148.207 92.7929L147.914 92.5H147.5V93.5ZM155.5 101.5L154.793 102.207L155.5 102.914L156.207 102.207L155.5 101.5ZM168 89L168.707 88.2929L168 87.5858L167.293 88.2929L168 89ZM180.5 101.5L179.793 102.207L180.5 102.914L181.207 102.207L180.5 101.5ZM193 89L193.707 89.7071L194.414 89L193.707 88.2929L193 89ZM180.5 76.5L181.207 75.7929L180.914 75.5H180.5V76.5ZM130.984 76.5L130.021 76.7673L130.224 77.5H130.984V76.5ZM129.407 93.1608C125.655 103.565 115.694 111 104 111V113C116.564 113 127.259 105.011 131.288 93.8392L129.407 93.1608ZM147.5 92.5H130.347V94.5H147.5V92.5ZM156.207 100.793L148.207 92.7929L146.793 94.2071L154.793 102.207L156.207 100.793ZM167.293 88.2929L154.793 100.793L156.207 102.207L168.707 89.7071L167.293 88.2929ZM181.207 100.793L168.707 88.2929L167.293 89.7071L179.793 102.207L181.207 100.793ZM192.293 88.2929L179.793 100.793L181.207 102.207L193.707 89.7071L192.293 88.2929ZM179.793 77.2071L192.293 89.7071L193.707 88.2929L181.207 75.7929L179.793 77.2071ZM130.984 77.5H180.5V75.5H130.984V77.5ZM104 57C116.405 57 126.859 65.3669 130.021 76.7673L131.948 76.2327C128.552 63.9889 117.327 55 104 55V57ZM77 84C77 69.0883 89.0883 57 104 57V55C87.9837 55 75 67.9837 75 84H77ZM104 111C89.0883 111 77 98.9117 77 84H75C75 100.016 87.9837 113 104 113V111ZM117 84C117 91.1797 111.18 97 104 97V99C112.284 99 119 92.2843 119 84H117ZM104 71C111.18 71 117 76.8203 117 84H119C119 75.7157 112.284 69 104 69V71ZM91 84C91 76.8203 96.8203 71 104 71V69C95.7157 69 89 75.7157 89 84H91ZM104 97C96.8203 97 91 91.1797 91 84H89C89 92.2843 95.7157 99 104 99V97Z", fill: "#00404D" }),
            React__namespace.createElement("path", { d: "M138.5 57V66", stroke: "#00404D", strokeWidth: "2" }),
            React__namespace.createElement("path", { d: "M139 104V124", stroke: "#00404D", strokeWidth: "2" }),
            React__namespace.createElement("path", { d: "M134 61.5L143 61.5", stroke: "#00404D", strokeWidth: "2" }),
            React__namespace.createElement("path", { d: "M129 114L149 114", stroke: "#00404D", strokeWidth: "2" }))));
};

/**
 * @internal For internal Amplify UI use only. May be removed in a future release.
 */
const AlertIcon = ({ variation, ariaHidden, ariaLabel, role, }) => {
    const icons = useIcons('alert');
    let icon;
    switch (variation) {
        case 'info':
            icon = icons?.info ?? (React__namespace.createElement(IconInfo, { "aria-hidden": ariaHidden, "aria-label": ariaLabel, role: role }));
            break;
        case 'error':
            icon = icons?.error ?? (React__namespace.createElement(IconError, { "aria-hidden": ariaHidden, "aria-label": ariaLabel, role: role }));
            break;
        case 'warning':
            icon = icons?.warning ?? (React__namespace.createElement(IconWarning, { "aria-hidden": ariaHidden, "aria-label": ariaLabel, role: role }));
            break;
        case 'success':
            icon = icons?.success ?? (React__namespace.createElement(IconCheckCircle, { "aria-hidden": ariaHidden, "aria-label": ariaLabel, role: role }));
            break;
    }
    return icon ? (React__namespace.createElement("span", { className: ui.ComponentClassName.AlertIcon }, icon)) : null;
};
AlertIcon.displayName = 'AlertIcon';

// Source: https://github.com/radix-ui/primitives/blob/7ae63b6cce6ea53ea5d65b6d411894c004b38f47/packages/react/use-layout-effect/src/useLayoutEffect.tsx
/**
 * On the server, React emits a warning when calling `useLayoutEffect`.
 * This is because neither `useLayoutEffect` nor `useEffect` run on the server.
 * We use this safe version which suppresses the warning by replacing it with a noop on the server.
 *
 * See: https://reactjs.org/docs/hooks-reference.html#uselayouteffect
 */
const useLayoutEffect = globalThis?.document ? React__namespace.useLayoutEffect : () => { };

// Adapted from https://github.com/radix-ui/primitives/blob/main/packages/react/id/src/id.tsx#L8
// Prefixed autogenerated id created by useStableId
const AUTO_GENERATED_ID_PREFIX = 'amplify-id';
// Create a local version of React.useId which will reference React.useId for React 18
// and fallback to noop for React 17 and below
// Note: We use `toString()` to prevent bundlers from trying to `import { useId } from 'react';`
// since it doesn't exist in React 17 and below (prevents https://github.com/aws-amplify/amplify-ui/issues/1154)
const useReactId = 
// disable eslint below to allow usage of casting React to `any`, which ensures that TS
// does not get confused about the existence of `useId` in React 17 and below
// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
React__namespace['useId'.toString()] || (() => undefined);
let count = 0;
/**
 * Create a uuid to use with amplify fields unless
 * an id is provided
 * @param id user specified id
 * @returns string
 */
const useStableId = (id) => {
    const [stableId, setStableId] = React__namespace.useState(useReactId());
    // React versions older than 18 will have client-side ids only
    useLayoutEffect(() => {
        if (!id) {
            setStableId((reactId) => reactId ?? String(count++));
        }
    }, [id]);
    return id ?? (stableId ? `${AUTO_GENERATED_ID_PREFIX}-${stableId}` : '');
};

// `FileSelect` input `type` must always be set to `file`
const INPUT_TYPE = 'file';
const TEST_ID = 'amplify-file-select';
const DEFAULT_PROPS = {
    style: { display: 'none' },
    type: 'file',
    'data-testid': TEST_ID,
};
/**
 * @internal @unstable
 */
const FileSelect = React__namespace["default"].forwardRef(function FileSelect({ multiple = true, type = 'FILE', testId = 'amplify-file-select', ...props }, ref) {
    return (React__namespace["default"].createElement("input", { ...DEFAULT_PROPS, ...props, ...(type === 'FOLDER' ? { webkitdirectory: '' } : undefined), "data-testid": testId, multiple: multiple, ref: ref, type: INPUT_TYPE }));
});
/**
 * @internal @unstable
 *
 * @usage
 * ```tsx
 *  function MyUploadButton() {
 *    const [files, setFiles] = React.useState<File[]>([]);
 *    const [fileSelect, handleFileSelect] = useFileSelect(setFiles);
 *    return (
 *      <>
 *        {fileSelect}
 *        <Button
 *          onClick={() => {
 *            handleFileSelect('file');
 *          }}
 *        />
 *      </>
 *    );
 *  }
 * ```
 */
const useFileSelect = (onSelect) => {
    const [inputProps, setInputProps] = React__namespace["default"].useState(undefined);
    const id = useStableId();
    const ref = React__namespace["default"].useRef(null);
    React__namespace["default"].useEffect(() => {
        if (inputProps) {
            ref.current?.click();
        }
    }, [id, inputProps]);
    const handleFileSelect = React__namespace["default"].useCallback((type, options) => {
        if (id !== ref.current?.id)
            return;
        setInputProps({ type, ...options });
    }, [id]);
    const fileSelect = (React__namespace["default"].createElement(FileSelect, { ...inputProps, id: id, onChange: (event) => {
            if (ui.isFunction(inputProps?.onChange))
                inputProps?.onChange(event);
            if (ui.isFunction(onSelect) && !!event.target.files?.length) {
                onSelect?.([...event.target.files]);
            }
            // Reset the input value to allow re-selecting the same file
            if (ref.current) {
                ref.current.value = '';
            }
            // clean up
            setInputProps(undefined);
        }, ref: ref }));
    return [fileSelect, handleFileSelect];
};

const FieldGroupIconPrimitive = ({ className, children, isVisible = true, excludeFromTabOrder = false, ...rest }, ref) => {
    return isVisible ? (React__namespace.createElement(View, { className: ui.classNames(ui.ComponentClassName.FieldGroupIcon, className), ref: ref, tabIndex: excludeFromTabOrder ? -1 : undefined, ...rest }, children)) : null;
};
const FieldGroupIcon = primitiveWithForwardRef(FieldGroupIconPrimitive);
FieldGroupIcon.displayName = 'FieldGroupIcon';

const FieldsetContext = React__namespace.createContext({
    isFieldsetDisabled: false,
});
/**
 * @description Fieldsets in HTML can be disabled, which disables all child
 * fieldsets and input controls. `useFieldset` passes the disabled state down
 * via context.
 */
const useFieldset = (fallback) => {
    const ctx = React__namespace.useContext(FieldsetContext);
    let value = ctx.isFieldsetDisabled;
    // cannot use nullish `??` since isFieldsetDisabled can be `false`
    if (value === undefined || value === false) {
        value = fallback;
    }
    return { isFieldsetDisabled: value };
};

const FlexPrimitive = ({ className, children, ...rest }, ref) => (React__namespace.createElement(View, { className: ui.classNames(ui.ComponentClassName.Flex, className), ref: ref, ...rest }, children));
/**
 * [📖 Docs](https://ui.docs.amplify.aws/react/components/flex)
 */
const Flex = primitiveWithForwardRef(FlexPrimitive);
Flex.displayName = 'Flex';

const LINEAR_EMPTY = 'linear-empty';
const LINEAR_FILLED = 'linear-filled';
const CIRCULAR_EMPTY = 'circular-empty';
const CIRCULAR_FILLED = 'circular-filled';
// radius + strokeWidth = 50
const CIRCULAR_STROKE_WIDTH = 8;
const RADIUS = 42;
// circumference = 2 * r * PI  (r = 42)
const CIRCUMFERENCE = 2 * RADIUS * Math.PI;
const LoaderPrimitive = ({ className, filledColor, emptyColor, size, variation, isDeterminate = false, isPercentageTextHidden = false, percentage = 0, ...rest }, ref) => {
    percentage = Math.min(percentage, 100);
    percentage = Math.max(percentage, 0);
    const percent = `${percentage}%`;
    const componentClasses = ui.classNames(ui.ComponentClassName.Loader, ui.classNameModifier(ui.ComponentClassName.Loader, size), ui.classNameModifier(ui.ComponentClassName.Loader, variation), ui.classNameModifierByFlag(ui.ComponentClassName.Loader, 'determinate', isDeterminate), className);
    const linearLoader = (React__namespace.createElement("g", null,
        React__namespace.createElement("line", { x1: "0", x2: "100%", y1: "50%", y2: "50%", style: { stroke: String(emptyColor) }, "data-testid": LINEAR_EMPTY }),
        React__namespace.createElement("line", { x1: "0", x2: isDeterminate ? percent : '100%', y1: "50%", y2: "50%", style: {
                // To get rid of the visible stroke linecap when percentage is 0
                stroke: isDeterminate && percentage === 0
                    ? 'none'
                    : filledColor
                        ? String(filledColor)
                        : undefined,
            }, "data-testid": LINEAR_FILLED }),
        isDeterminate ? (React__namespace.createElement("text", { "aria-live": "polite", className: ui.classNames(ui.ComponentClassName.LoaderLabel, isPercentageTextHidden ? ui.ComponentClassName.VisuallyHidden : null), 
            // -1% offset makes the text position look nicest
            x: `${ -1 + percentage}%`, y: "200%" }, percent)) : null));
    // r + stroke-width should add up to 50% to avoid overflow
    const circularLoader = (React__namespace.createElement("g", null,
        React__namespace.createElement("circle", { cx: "50%", cy: "50%", r: `${RADIUS}%`, strokeWidth: `${CIRCULAR_STROKE_WIDTH}%`, style: { stroke: String(emptyColor) }, "data-testid": CIRCULAR_EMPTY }),
        React__namespace.createElement("circle", { cx: "50%", cy: "50%", r: `${RADIUS}%`, strokeWidth: `${CIRCULAR_STROKE_WIDTH}%`, style: {
                stroke: String(filledColor),
                strokeDasharray: isDeterminate
                    ? `${CIRCUMFERENCE}% ${CIRCUMFERENCE}%`
                    : undefined,
                strokeDashoffset: isDeterminate
                    ? `${CIRCUMFERENCE - (CIRCUMFERENCE * percentage) / 100}%`
                    : undefined,
            }, "data-testid": CIRCULAR_FILLED }),
        isDeterminate ? (React__namespace.createElement("text", { "aria-live": "polite", className: ui.classNames(ui.ComponentClassName.LoaderLabel, isPercentageTextHidden ? ui.ComponentClassName.VisuallyHidden : null), 
            // this x and y make text position look nicest
            x: "130%", y: "80%" }, percent)) : null));
    return (React__namespace.createElement(View, { as: "svg", "aria-valuenow": isDeterminate ? percentage : undefined, className: componentClasses, ref: ref, role: "progressbar", ...rest }, variation === 'linear' ? linearLoader : circularLoader));
};
/**
 * [📖 Docs](https://ui.docs.amplify.aws/react/components/loader)
 */
const Loader = primitiveWithForwardRef(LoaderPrimitive);
Loader.displayName = 'Loader';

// These variations support colorThemes. 'undefined' accounts for our
// 'default' variation which is not named.
const supportedVariations = ['link', 'primary', undefined];
const ButtonPrimitive = ({ className, children, colorTheme, isFullWidth = false, isDisabled, isLoading, loadingText = '', size, type = 'button', variation, ...rest }, ref) => {
    // Creates our colorTheme modifier string based on if the variation
    // supports colorThemes and a colorTheme is used.
    const colorThemeModifier = supportedVariations.includes(variation) && colorTheme
        ? `${variation ?? 'outlined'}--${colorTheme}`
        : undefined;
    const { isFieldsetDisabled: shouldBeDisabled } = useFieldset(isDisabled ?? isLoading ?? rest['disabled']);
    const componentClasses = ui.classNames(ui.ComponentClassName.Button, ui.ComponentClassName.FieldGroupControl, ui.classNameModifier(ui.ComponentClassName.Button, variation), ui.classNameModifier(ui.ComponentClassName.Button, colorThemeModifier), ui.classNameModifier(ui.ComponentClassName.Button, size), ui.classNameModifierByFlag(ui.ComponentClassName.Button, 'disabled', shouldBeDisabled), ui.classNameModifierByFlag(ui.ComponentClassName.Button, 'loading', isLoading), ui.classNameModifierByFlag(ui.ComponentClassName.Button, 'fullwidth', isFullWidth), className);
    return (React__namespace.createElement(View, { ref: ref, as: "button", className: componentClasses, isDisabled: shouldBeDisabled, type: type, ...rest }, isLoading ? (React__namespace.createElement(Flex, { as: "span", className: ui.ComponentClassName.ButtonLoaderWrapper },
        React__namespace.createElement(Loader, { size: size }),
        loadingText ? loadingText : null)) : (children)));
};
/**
 * [📖 Docs](https://ui.docs.amplify.aws/react/components/button)
 */
const Button = primitiveWithForwardRef(ButtonPrimitive);
Button.displayName = 'Button';

const FieldGroupIconButtonPrimitive = ({ children, className, ...rest }, ref) => (React__namespace.createElement(FieldGroupIcon, { as: Button, className: ui.classNames(ui.ComponentClassName.FieldGroupIconButton, className), ref: ref, ...rest }, children));
const FieldGroupIconButton = primitiveWithForwardRef(FieldGroupIconButtonPrimitive);
FieldGroupIconButton.displayName = 'FieldGroupIconButton';

const ariaLabelText = ComponentText.Fields.clearButtonLabel;
const FieldClearButtonPrimitive = ({ ariaLabel = ariaLabelText, size, ...rest }, ref) => {
    const icons = useIcons('field');
    return (React__namespace.createElement(FieldGroupIconButton, { ariaLabel: ariaLabel, size: size, ref: ref, ...rest }, icons?.clear ?? React__namespace.createElement(IconClose, null)));
};
const FieldClearButton = primitiveWithForwardRef(FieldClearButtonPrimitive);
FieldClearButton.displayName = 'FieldClearButton';

const TextPrimitive = ({ as = 'p', className, children, isTruncated, variation, ...rest }, ref) => {
    const componentClasses = ui.classNames(ui.ComponentClassName.Text, ui.classNameModifier(ui.ComponentClassName.Text, variation), ui.classNameModifierByFlag(ui.ComponentClassName.Text, 'truncated', isTruncated), className);
    return (React__namespace.createElement(View, { as: as, className: componentClasses, ref: ref, ...rest }, children));
};
/**
 * [📖 Docs](https://ui.docs.amplify.aws/react/components/text)
 */
const Text = primitiveWithForwardRef(TextPrimitive);
Text.displayName = 'Text';

const QA_FIELD_DESCRIPTION = 'qa-field-description';
const FieldDescription = ({ descriptiveText, labelHidden, ...rest }) => descriptiveText ? (React__namespace.createElement(Text, { "data-testid": QA_FIELD_DESCRIPTION, className: ui.classNames(ui.ComponentClassName.FieldDescription, {
        [ui.ComponentClassName.VisuallyHidden]: labelHidden,
    }), ...rest }, descriptiveText)) : null;
FieldDescription.displayName = 'FieldDescription';

const FieldErrorMessage = ({ errorMessage, hasError, ...rest }) => {
    return hasError && errorMessage ? (React__namespace.createElement(Text, { className: ui.ComponentClassName.FieldErrorMessage, ...rest }, errorMessage)) : null;
};
FieldErrorMessage.displayName = 'FieldErrorMessage';

const LabelPrimitive = ({ children, className, visuallyHidden, ...rest }, ref) => {
    return (React__namespace.createElement(View, { as: "label", className: ui.classNames(ui.ComponentClassName.Label, className, {
            [ui.ComponentClassName.VisuallyHidden]: visuallyHidden,
        }), ref: ref, ...rest }, children));
};
const Label = primitiveWithForwardRef(LabelPrimitive);
Label.displayName = 'Label';

const FieldPrimitive = (props, ref) => {
    const { className, size, testId, children, label, labelHidden, errorMessage, hasError, descriptiveText, ...rest } = props;
    return (React__namespace.createElement(Flex, { className: ui.classNames(ui.ComponentClassName.Field, ui.classNameModifier(ui.ComponentClassName.Field, size), className), testId: testId, ref: ref, ...rest },
        label ? React__namespace.createElement(Label, { visuallyHidden: labelHidden }, label) : null,
        React__namespace.createElement(FieldDescription, { labelHidden: labelHidden, descriptiveText: descriptiveText }),
        children,
        errorMessage ? (React__namespace.createElement(FieldErrorMessage, { hasError: hasError, errorMessage: errorMessage })) : null));
};
const Field = primitiveWithForwardRef(FieldPrimitive);
Field.displayName = 'Field';

exports.ARROW_DOWN = ARROW_DOWN;
exports.ARROW_UP = ARROW_UP;
exports.AlertIcon = AlertIcon;
exports.Button = Button;
exports.ComponentPropsToStylePropsMap = ComponentPropsToStylePropsMap;
exports.ComponentPropsToStylePropsMapKeys = ComponentPropsToStylePropsMapKeys;
exports.ComponentText = ComponentText;
exports.ENTER_KEY = ENTER_KEY;
exports.ESCAPE_KEY = ESCAPE_KEY;
exports.Field = Field;
exports.FieldClearButton = FieldClearButton;
exports.FieldDescription = FieldDescription;
exports.FieldErrorMessage = FieldErrorMessage;
exports.FieldGroupIcon = FieldGroupIcon;
exports.FieldGroupIconButton = FieldGroupIconButton;
exports.FieldsetContext = FieldsetContext;
exports.FileSelect = FileSelect;
exports.Flex = Flex;
exports.Icon = Icon;
exports.IconAdd = IconAdd;
exports.IconCheck = IconCheck;
exports.IconCheckCircle = IconCheckCircle;
exports.IconCheckCircleFill = IconCheckCircleFill;
exports.IconChevronLeft = IconChevronLeft;
exports.IconChevronRight = IconChevronRight;
exports.IconClose = IconClose;
exports.IconError = IconError;
exports.IconExpandMore = IconExpandMore;
exports.IconIndeterminate = IconIndeterminate;
exports.IconInfo = IconInfo;
exports.IconMenu = IconMenu;
exports.IconPasskey = IconPasskey;
exports.IconRemove = IconRemove;
exports.IconSearch = IconSearch;
exports.IconStar = IconStar;
exports.IconUser = IconUser;
exports.IconVisibility = IconVisibility;
exports.IconVisibilityOff = IconVisibilityOff;
exports.IconWarning = IconWarning;
exports.IconsContext = IconsContext;
exports.Label = Label;
exports.Loader = Loader;
exports.Text = Text;
exports.ThemeContext = ThemeContext;
exports.View = View;
exports.getConsecutiveIntArray = getConsecutiveIntArray;
exports.getStyleValue = getStyleValue;
exports.getValueAtCurrentBreakpoint = getValueAtCurrentBreakpoint;
exports.primitiveWithForwardRef = primitiveWithForwardRef;
exports.strHasLength = strHasLength;
exports.useAuth = useAuth;
exports.useBreakpoint = useBreakpoint;
exports.useColorMode = useColorMode;
exports.useDeprecationWarning = useDeprecationWarning;
exports.useFieldset = useFieldset;
exports.useFileSelect = useFileSelect;
exports.useIcons = useIcons;
exports.useStableId = useStableId;
exports.useStyles = useStyles;
exports.useTheme = useTheme;
