// react-cookie-consent/dist/react-cookie-consent.es.js
import a, { Component as G } from "react";
/*! js-cookie v3.0.5 | MIT */
function y(t) {
  for (var e = 1; e < arguments.length; e++) {
    var n = arguments[e];
    for (var l in n)
      t[l] = n[l];
  }
  return t;
}
var J = {
  read: function(t) {
    return t[0] === '"' && (t = t.slice(1, -1)), t.replace(/(%[\dA-F]{2})+/gi, decodeURIComponent);
  },
  write: function(t) {
    return encodeURIComponent(t).replace(
      /%(2[346BF]|3[AC-F]|40|5[BDE]|60|7[BCD])/g,
      decodeURIComponent
    );
  }
};
function S(t, e) {
  function n(o, i, s) {
    if (!(typeof document > "u")) {
      s = y({}, e, s), typeof s.expires == "number" && (s.expires = new Date(Date.now() + s.expires * 864e5)), s.expires && (s.expires = s.expires.toUTCString()), o = encodeURIComponent(o).replace(/%(2[346B]|5E|60|7C)/g, decodeURIComponent).replace(/[()]/g, escape);
      var c = "";
      for (var r in s)
        s[r] && (c += "; " + r, s[r] !== !0 && (c += "=" + s[r].split(";")[0]));
      return document.cookie = o + "=" + t.write(i, o) + c;
    }
  }
  function l(o) {
    if (!(typeof document > "u" || arguments.length && !o)) {
      for (var i = document.cookie ? document.cookie.split("; ") : [], s = {}, c = 0; c < i.length; c++) {
        var r = i[c].split("="), m = r.slice(1).join("=");
        try {
          var u = decodeURIComponent(r[0]);
          if (s[u] = t.read(m, u), o === u)
            break;
        } catch {
        }
      }
      return o ? s[o] : s;
    }
  }
  return Object.create(
    {
      set: n,
      get: l,
      remove: function(o, i) {
        n(
          o,
          "",
          y({}, i, {
            expires: -1
          })
        );
      },
      withAttributes: function(o) {
        return S(this.converter, y({}, this.attributes, o));
      },
      withConverter: function(o) {
        return S(y({}, this.converter, o), this.attributes);
      }
    },
    {
      attributes: { value: Object.freeze(e) },
      converter: { value: Object.freeze(t) }
    }
  );
}
var b = S(J, { path: "/" });
const Q = ({ condition: t, wrapper: e, children: n }) => t ? e(n) : n, B = {
  TOP: "top",
  BOTTOM: "bottom",
  NONE: "none"
};
var I = /* @__PURE__ */ ((t) => (t.STRICT = "strict", t.LAX = "lax", t.NONE = "none", t))(I || {});
const x = {
  HIDDEN: "hidden",
  SHOW: "show",
  BY_COOKIE_VALUE: "byCookieValue"
}, D = "CookieConsent", Z = ({
  children: t,
  ...e
}) => /* @__PURE__ */ a.createElement("button", { ...e }, t), d = {
  disableStyles: !1,
  hideOnAccept: !0,
  hideOnDecline: !0,
  location: B.BOTTOM,
  visible: x.BY_COOKIE_VALUE,
  onAccept: (t) => {
  },
  onDecline: () => {
  },
  cookieName: D,
  cookieValue: "true",
  declineCookieValue: "false",
  setDeclineCookie: !0,
  buttonText: "I understand",
  declineButtonText: "I decline",
  debug: !1,
  expires: 365,
  containerClasses: "CookieConsent",
  contentClasses: "",
  buttonClasses: "",
  buttonWrapperClasses: "",
  declineButtonClasses: "",
  buttonId: "rcc-confirm-button",
  declineButtonId: "rcc-decline-button",
  extraCookieOptions: {},
  disableButtonStyles: !1,
  enableDeclineButton: !1,
  flipButtons: !1,
  sameSite: I.LAX,
  ButtonComponent: Z,
  overlay: !1,
  overlayClasses: "",
  onOverlayClick: () => {
  },
  acceptOnOverlayClick: !1,
  ariaAcceptLabel: "Accept cookies",
  ariaDeclineLabel: "Decline cookies",
  acceptOnScroll: !1,
  acceptOnScrollPercentage: 25,
  customContentAttributes: {},
  customContainerAttributes: {},
  customButtonProps: {},
  customDeclineButtonProps: {},
  customButtonWrapperAttributes: {},
  style: {},
  buttonStyle: {},
  declineButtonStyle: {},
  contentStyle: {},
  overlayStyle: {}
}, ee = {
  visible: !1,
  style: {
    alignItems: "baseline",
    background: "#353535",
    color: "white",
    display: "flex",
    flexWrap: "wrap",
    justifyContent: "space-between",
    left: "0",
    position: "fixed",
    width: "100%",
    zIndex: "999"
  },
  buttonStyle: {
    background: "#ffd42d",
    border: "0",
    borderRadius: "0px",
    boxShadow: "none",
    color: "black",
    cursor: "pointer",
    flex: "0 0 auto",
    padding: "5px 10px",
    margin: "15px"
  },
  declineButtonStyle: {
    background: "#c12a2a",
    border: "0",
    borderRadius: "0px",
    boxShadow: "none",
    color: "#e5e5e5",
    cursor: "pointer",
    flex: "0 0 auto",
    padding: "5px 10px",
    margin: "15px"
  },
  contentStyle: {
    flex: "1 0 300px",
    margin: "15px"
  },
  overlayStyle: {
    position: "fixed",
    left: 0,
    top: 0,
    width: "100%",
    height: "100%",
    zIndex: "999",
    backgroundColor: "rgba(0,0,0,0.3)"
  }
}, te = (t = D) => {
  const e = b.get(t);
  return e === void 0 ? b.get(w(t)) : e;
}, ne = (t = D) => {
  b.remove(t);
}, w = (t) => `${t}-legacy`, T = class T extends G {
  constructor() {
    super(...arguments), this.state = ee, this.handleScroll = () => {
      const { acceptOnScrollPercentage: e } = { ...d, ...this.props }, n = document.documentElement, l = document.body, o = "scrollTop", i = "scrollHeight";
      (n[o] || l[o]) / ((n[i] || l[i]) - n.clientHeight) * 100 > e && this.accept(!0);
    }, this.removeScrollListener = () => {
      const { acceptOnScroll: e } = this.props;
      e && globalThis.removeEventListener("scroll", this.handleScroll);
    };
  }
  componentDidMount() {
    const { debug: e } = this.props;
    (this.getCookieValue() === void 0 || e) && (this.setState({ visible: !0 }), this.props.acceptOnScroll && globalThis.addEventListener("scroll", this.handleScroll, { passive: !0 }));
  }
  componentWillUnmount() {
    this.removeScrollListener();
  }
  /**
   * Set a persistent accept cookie
   */
  accept(e = !1) {
    const { cookieName: n, cookieValue: l, hideOnAccept: o, onAccept: i } = {
      ...d,
      ...this.props
    };
    this.setCookie(n, l), i(e ?? !1), o && (this.setState({ visible: !1 }), this.removeScrollListener());
  }
  /**
   * Handle a click on the overlay
   */
  overlayClick() {
    const { acceptOnOverlayClick: e, onOverlayClick: n } = {
      ...d,
      ...this.props
    };
    e && this.accept(), n();
  }
  /**
   * Set a persistent decline cookie
   */
  decline() {
    const { cookieName: e, declineCookieValue: n, hideOnDecline: l, onDecline: o, setDeclineCookie: i } = {
      ...d,
      ...this.props
    };
    i && this.setCookie(e, n), o(), l && this.setState({ visible: !1 });
  }
  /**
   * Function to set the consent cookie based on the provided variables
   * Sets two cookies to handle incompatible browsers, more details:
   * https://web.dev/samesite-cookie-recipes/#handling-incompatible-clients
   */
  setCookie(e, n) {
    const { extraCookieOptions: l, expires: o, sameSite: i } = this.props;
    let { cookieSecurity: s } = this.props;
    s === void 0 && (s = globalThis.location ? globalThis.location.protocol === "https:" : !0);
    const c = { expires: o, ...l, sameSite: i, secure: s };
    i === I.NONE && b.set(w(e), n, c), b.set(e, n, c);
  }
  /**
   * Returns the value of the consent cookie
   * Retrieves the regular value first and if not found the legacy one according
   * to: https://web.dev/samesite-cookie-recipes/#handling-incompatible-clients
   */
  getCookieValue() {
    const { cookieName: e } = this.props;
    return te(e);
  }
  render() {
    switch (this.props.visible) {
      case x.HIDDEN:
        return null;
      case x.BY_COOKIE_VALUE:
        if (!this.state.visible)
          return null;
        break;
    }
    const {
      location: e,
      style: n,
      buttonStyle: l,
      declineButtonStyle: o,
      contentStyle: i,
      disableStyles: s,
      buttonText: c,
      declineButtonText: r,
      containerClasses: m,
      contentClasses: u,
      buttonClasses: E,
      buttonWrapperClasses: L,
      declineButtonClasses: V,
      buttonId: P,
      declineButtonId: R,
      disableButtonStyles: U,
      enableDeclineButton: W,
      flipButtons: H,
      ButtonComponent: N,
      overlay: _,
      overlayClasses: z,
      overlayStyle: A,
      ariaAcceptLabel: K,
      ariaDeclineLabel: Y,
      customContainerAttributes: F,
      customContentAttributes: $,
      customButtonProps: M,
      customDeclineButtonProps: X,
      customButtonWrapperAttributes: q
    } = this.props;
    let p = {}, f = {}, h = {}, g = {}, O = {};
    switch (s ? (p = Object.assign({}, n), f = Object.assign({}, l), h = Object.assign({}, o), g = Object.assign({}, i), O = Object.assign({}, A)) : (p = Object.assign({}, { ...this.state.style, ...n }), g = Object.assign({}, { ...this.state.contentStyle, ...i }), O = Object.assign({}, { ...this.state.overlayStyle, ...A }), U ? (f = Object.assign({}, l), h = Object.assign({}, o)) : (f = Object.assign({}, { ...this.state.buttonStyle, ...l }), h = Object.assign(
      {},
      { ...this.state.declineButtonStyle, ...o }
    ))), e) {
      case B.TOP:
        p.top = "0";
        break;
      case B.BOTTOM:
        p.bottom = "0";
        break;
    }
    const C = [];
    return W && C.push(
      /* @__PURE__ */ a.createElement(
        N,
        {
          key: "declineButton",
          style: h,
          className: V,
          id: R,
          "aria-label": Y,
          onClick: () => {
            this.decline();
          },
          ...X
        },
        r
      )
    ), C.push(
      /* @__PURE__ */ a.createElement(
        N,
        {
          key: "acceptButton",
          style: f,
          className: E,
          id: P,
          "aria-label": K,
          onClick: () => {
            this.accept();
          },
          ...M
        },
        c
      )
    ), H && C.reverse(), /* @__PURE__ */ a.createElement(
      Q,
      {
        condition: _,
        wrapper: (v) => /* @__PURE__ */ a.createElement(
          "div",
          {
            style: O,
            className: z,
            onClick: () => {
              this.overlayClick();
            },
            onKeyDown: (k) => {
              (k.key === "Enter" || k.key === " ") && (k.preventDefault(), this.overlayClick());
            },
            role: "button",
            tabIndex: 0,
            "aria-label": "Close cookie consent banner"
          },
          v
        )
      },
      /* @__PURE__ */ a.createElement("div", { className: `${m}`, style: p, ...F }, /* @__PURE__ */ a.createElement("div", { style: g, className: u, ...$ }, this.props.children), /* @__PURE__ */ a.createElement("div", { className: `${L}`, ...q }, C.map((v) => v)))
    );
  }
};
T.defaultProps = d;
let j = T;
export {
  Q as ConditionalWrapper,
  j as CookieConsent,
  b as Cookies,
  B as OPTIONS,
  B as POSITION_OPTIONS,
  I as SAME_SITE_OPTIONS,
  x as VISIBILITY_OPTIONS,
  x as VISIBLE_OPTIONS,
  j as default,
  D as defaultCookieConsentName,
  te as getCookieConsentValue,
  w as getLegacyCookieName,
  ne as resetCookieConsentValue
};
